cmd/minikube/cmd/cp.go

```diff
@@ -59,6 +59,7 @@ var cpCmd = &cobra.Command{
 			out.ErrLn("%v", errors.Wrap(err, "getting file asset"))
 			os.Exit(1)
 		}
+		defer fa.Close()
```

Comment:
lets move the defer before the os.Exit() we should try to close even if there is an error..
```
		fa, err := assets.NewFileAsset(srcPath, pt.Dir(dstPath), pt.Base(dstPath), "0644")
		defer fa.Close()
		if err != nil {
			out.ErrLn("%v", errors.Wrap(err, "getting file asset"))
			os.Exit(1)
		}
```

Reply:
deferred functions are not called after `os.Exit`

Reply:
Good call ! I just noticed why were we doing os exit here ?
@daehyeok let's use the minikube'e exit package instead of calling os exit

Reply:
NewFileAsset always return nil for FileAsset if error occurred.

https://github.com/kubernetes/minikube/blob/78618d04fc06d06d4e60aaa947feef060bdae06a/pkg/minikube/assets/vm_assets.go#L94

Like `os.Open`, How about close FileAsset only when `NewFileAsset` success?

Reply:
I changed to use exit package except NewFileAsset. Which reason type should i use(or should i add new type?)
```Go
fa, err := assets.NewFileAsset(srcPath, pt.Dir(dstPath), pt.Base(dstPath), "0644")
if err != nil {
	out.ErrLn("%v", errors.Wrap(err, "getting file asset"))
	os.Exit(1)
}
```
and os.Exit is depreacted for cmd?? it looks bunch of os.Exit still exist in cmd package.

```
~/workspace/minikube/cmd/minikube/cmd❯ rg os\.Exit
ssh.go
64:                     os.Exit(1)

logs.go
105:                    os.Exit(reason.ExSvcError)

mount.go
76:                     os.Exit(0)

status.go
290:                    os.Exit(exitCode(statuses))

node_list.go
53:             os.Exit(0)

root.go
157:            defer os.Exit(reason.ExProgramUsage)

ssh-host.go
72:                     os.Exit(1)
79:                             os.Exit(1)
96:                             os.Exit(1)
103:                            os.Exit(1)

kubectl.go
79:                             os.Exit(1)
94:                     os.Exit(1)
105:                    os.Exit(1)
121:                    os.Exit(rc)

cp.go
74:                     os.Exit(1)

service_list.go
48:                     os.Exit(reason.ExSvcUnavailable)

start.go
304:            os.Exit(0)

node_start.go
55:                     os.Exit(0)

config/profile.go
41:                     os.Exit(0)
74:                     os.Exit(0)

config/profile_list.go
203:            os.Exit(reason.ExGuestError)
```

Reply:
Just a note: minikube's `exit` package uses the same `os.Exit()` inside, so deferred functions will be skipped anyway